Code for Admin Module (User Management)

chirag-compro opened the pull request.

Commit:
Code for admin module

Commit:
Code for administrtaion users

Commit:
Code for administrtaion users

Commit:
Code enhacements for user detail form

Commit:
ApI name changed as per review

Commit:
Removing Consoles

ashish-compro commented on client/app/_services/user.service.ts:
```diff
@@ -0,0 +1,83 @@
+  }
+  getUsers(): Observable<User[]>{
+    if(this.userData && this.useradded == false){
+      return Observable.of(this.userData);
```
It should call the server to get the updated data always when the component of user list is displayed.

ashish-compro commented on client/app/_services/user.service.ts:
```diff
@@ -0,0 +1,83 @@
+import { User } from './userModel';
+@Injectable()
+export class UserService {
+public userData: any;
```
According to me this data should either be moved to a data store or should be fetched from the server every time.

Commit:
code enhancement as per review

chirag-compro commented on client/app/_services/user.service.ts:
```diff
@@ -0,0 +1,75 @@
+    let body = res.json();
+    return body || { };
+  }
+  private handleError (error: Response | any) {
```
error handler can be a common service for all the modules.

chirag-compro commented on client/app/administration/user-details-form/user-details-form.component.ts:
```diff
@@ -0,0 +1,46 @@
+  }
+
+  ngOnInit() {
+    this.instance = jQuery('.parsleyjs').parsley();
```
rename instance  to ParsleyForm

chirag-compro commented on client/app/administration/user-list/user-list.component.ts:
```diff
@@ -0,0 +1,67 @@
+                     .subscribe(
+                       Data => this.Data = Data,
+                       error =>  this.errorMessage = <any>error);
+					   this.userData = this.Data["userData"];
```
this.Data = Data, can be replaced with this.userData = this.Data["userData"];

Commit:
Code for UI pop ups and enhacement as per review.
1 File changesd for taskData service to make taskId comparison case insensitive

chirag-compro commented on client/app/administration/add-user/add-user.component.html:
```diff
@@ -0,0 +1,9 @@
+<section class="widget" widget>
+  <div class="row">
+    <div class="col-lg-6 offset-lg-3 col-xs-12">
+      <app-user-details-form [title]="'Add the information about the user you want to add in Sims Builder.'" [mode]="'add'" (closeFormEvent)="closeFormEventListener($event)"></app-user-details-form>
```
closeFormEventListener can be renamed to closeFormListener

chirag-compro commented on client/app/administration/user-details-form/user-details-form.component.ts:
```diff
@@ -0,0 +1,64 @@
+@Input() title:string;
+@Input() user;
+@Input() mode;
+@Output() closeFormEvent: EventEmitter<any> = new EventEmitter();
```
decide a better name for this event closeFormEvent and send message with each emit

chirag-compro commented on client/app/administration/add-user/add-user.component.ts:
```diff
@@ -0,0 +1,25 @@
+							theme: 'block'}
+   }
+  closeFormEventListener(message){
+    if(message){
```
Map of message emited and message shown can be implemented.

tanujaggarwal commented on server/routes/index.js:
```diff
@@ -4,13 +4,14 @@ var loginRouter = require('./login/login.route');
 var skillRoutes = require('./skill.routes');
 var verifyToken = require('./login/verifyToken');
-
+var userData = require('./userData');
```
rename to user

tanujaggarwal commented on server/routes/userData.js:
```diff
@@ -0,0 +1,99 @@
+const express = require('express');
+const router = express.Router();
+const fs = require('fs');
```
delete this line

tanujaggarwal commented on server/routes/userData.js:
```diff
@@ -0,0 +1,99 @@
+})
+router.put("/",function(req,res){
+	var user = req.body;
+	var userUpdateData = {  $set:{
```
we can set parameters if only present in the request
request should contain the parameters modified only.

tanujaggarwal commented on server/routes/index.js:
```diff
@@ -4,13 +4,14 @@ var loginRouter = require('./login/login.route');
 	router.use('*', verifyToken);
 	router.use('/skill', skillRoutes);
   router.use('/fetchTaskData', taskBuilderData);	
+	router.use('/user', userData);	
```
If all users should be allowed to access this API or only admin users should be allowed
Baban sir to confirm.

Commit:
Code changes as per review

vsingb3 commented on client/app/administration/add-user/add-user.component.ts:
```diff
@@ -0,0 +1,26 @@
+							theme: 'block'}
+   }
+  emittedEventListener(message){
+    if(Map[message]!= "No Changes"){
```
All the valid messages should be declared in an enum

vsingb3 commented on client/app/home/navbar/navbar.component.ts:
```diff
@@ -31,11 +38,27 @@ export class Navbar implements OnInit {
     this.router.navigate(['/login']);
   }
-
+  checkUsertype(userRole){
```
IsAdminUser

Commit:
Code changes as per review

chirag-compro committed:
Merge branch 'develop' into Administration

Commit:
Renaming dropdownModule (changes in ng2-bootstrap npm module)

Commit:
changing version for npm module

Commit:
Code enhancement
